ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -78,65 +78,87 @@ bool PowerSystemNode::loadSystemConfig()
   return true;
 }
 
-PrognoserVector PowerSystemNode::loadPowerProfile(const string& filename)
+PrognoserVector PowerSystemNode::loadPowerProfile(const string& filename, string custom_file)
 {
   ifstream file(filename);
+
   if (file.fail())
   {
-    cerr << "Unable to open data file" << filename << endl;
+    ROS_WARN_STREAM("Could not find a custom file in the 'profiles' directory with name '"
+                          << custom_file << "'. Deactivate fault and try again.");
+    return PrognoserVector();
   }
+
   // Skip header line
   file.ignore(numeric_limits<streamsize>::max(), '\n');
 
   auto now = system_clock::now();
 
   PrognoserVector result;
-  while (file.good())
+
+  // Line number starts at 2 instead of 1 because the first line is the header.
+  int line_number = 2;
```

Comment:
This will need to be a user requirement to work.   Ideally, the line number would start at the first line of content, however this is fine for this Jira.

Reply:
This is based primarily off of me reading files in VS code; it labels each line in the file with a number and starting at 2 means the line number provided in an error message is consistent with the total number of lines in the entire file. I'll leave it as is, but whether or not this starts at 2, 1, or even 0 is primarily up to preference. It can easily be changed later anyways.